cargo-concordium/src/context.rs

```diff
@@ -0,0 +1,141 @@
+use anyhow::anyhow;
+use concordium_contracts_common::{
+    AccountAddress, Address, Amount, ContractAddress, OwnedPolicy, SlotTime,
+};
+use wasm_chain_integration::{ExecResult, HasChainMetadata, HasInitContext, HasReceiveContext};
+
+/// A chain metadata with an optional field.
+/// Used when simulating contracts to allow the user to only specify the
+/// necessary context fields.
+#[derive(serde::Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub(crate) struct ChainMetadataOpt {
+    slot_time: Option<SlotTime>,
+}
+
+impl ChainMetadataOpt {
+    fn new() -> Self {
+        Self {
+            slot_time: None,
+        }
+    }
+}
+
+impl Default for ChainMetadataOpt {
+    fn default() -> Self { Self::new() }
+}
```

Comment:
I think you should just derive Default for `ChainMetadataOpt`. It will have exactly this implementation (i.e., all fields are default value, which is None for options).